rocq-9.0v1 branch fails to build with OCaml 5.x: Unbound module Big_int

`rocq-9.0v1` branch fails to build with OCaml 5.x: Unbound module Big_int

## Description

The `rocq-9.0v1` branch fails to compile when using OCaml 5.x due to the `Big_int` module no longer being in the standard library.

### Environment
- **OCaml version:** 5.1.1
- **opam switch:** 5.1.1
- **Rocq version:** 9.0.1
- **OS:** Linux

### Steps to Reproduce

```bash
opam pin add coq-smtcoq git+https://github.com/smtcoq/smtcoq.git#rocq-9.0v1 -y
```

### Error Message

```
[ERROR] The compilation of coq-smtcoq.2.3+8.20 failed at "make -j3".

# command   make -j3
# exit-code 2

File "verit/veritParser.mly", line 46, characters 8-23:
Error: Unbound module Big_int
make[3]: *** [CoqMakefile:720: verit/veritParser.cmi] Error 2
```

### Root Cause

In OCaml 5.0, the `Big_int` module was removed from the standard library and is now provided by the `num` package. The `num` package is installed (version 1.6), but the SMTCoq code references `Big_int` directly without:
1. Using `open Num`, or
2. Qualifying as `Num.Big_int`

### Suggested Fix

In `verit/veritParser.mly` (and any other affected files), either:

**Option 1:** Add `open Num` at the top of the file
```ocaml
open Num
(* existing code using Big_int works unchanged *)
```

**Option 2:** Qualify Big_int references
```ocaml
(* Change *)
Big_int.big_int_of_string x
(* To *)
Num.Big_int.big_int_of_string x
```

**Option 3 (recommended long-term):** Migrate to Zarith
As suggested in [Coq PR #11742](https://github.com/rocq-prover/rocq/pull/11742), switching from `num`/`Big_int` to `Zarith` would be the modern approach.

### Workaround

Use OCaml 4.14 instead of 5.x:
```bash
opam switch create smtcoq-env 4.14.2
eval $(opam env)
# Then install Rocq 9.0 and SMTCoq
```

### Labels
- bug
- ocaml-5
- build

---

**Note:** This issue was identified while attempting to use SMTCoq with Rocq 9.0.1 on a system running OCaml 5.1.1.

--- a/src/CoqMakefile.local
+++ b/src/CoqMakefile.local
@@ -29,7 +29,8 @@ cleanall::
 
 CAMLLEX = $(CAMLBIN)ocamllex
 CAMLYACC = $(CAMLBIN)ocamlyacc
-CAMLPKGS += -package rocq-runtime.plugins.micromega
+# OCaml 5.x compatibility: num package provides Big_int module
+CAMLPKGS += -package num -package rocq-runtime.plugins.micromega
 
 merlin-hook::
 	$(HIDE)echo 'PKG num' >> .merlin

